### PR TITLE
VirtualCard: Support refund transactions

### PR DESCRIPTION
**Pull request by kewitz**
This covers the missing edge case of refunded virtual card charges.
Notice I'm only creating the transaction pair here, I don't think it makes sense to create an expense with a negative value.

### CODE REVIEW

File: server/paymentProviders/privacy/index.ts [hunk 1]
```diff
@@ -14,10 +14,10 @@ import VirtualCardModel from '../../models/VirtualCard';
 import { Transaction } from '../../types/privacy';
 import { CardProviderService } from '../types';
 
-const createExpense = async (
+const processTransaction = async (
   privacyTransaction: Transaction,
   opts?: { host?: any; collective?: any; hostCurrencyFxRate?: number },
-): Promise<any> => {
+): Promise<typeof models.Expense | undefined> => {
   const amount = privacyTransaction.settled_amount;
```
Comment (znarf) [inline server/paymentProviders/privacy/index.ts:21]: What is our guarantee that `settled_amount` is in USD here. Is it really always the case? Even if the charge is in another currency?

File: server/paymentProviders/privacy/index.ts [hunk 2]
```diff
@@ -125,7 +153,10 @@ const createExpense = async (
     return expense;
   });
 
-  if (collective.settings?.ignoreExpenseMissingReceiptAlerts !== true) {
+  if (
+    expense && // Expense is undefined if the the transaction is a refund
```
Comment (znarf) [inline server/paymentProviders/privacy/index.ts:157]: You have `isRefund` available as a variable, maybe it would be more explicit to use it?

**Review by znarf — APPROVED** [review]
I'm good with the change.

@Betree have a look so you have the context about this behavior.

File: server/paymentProviders/privacy/index.ts (hunk 1 above)
Comment (kewitz) [reply server/paymentProviders/privacy/index.ts:21]: Yes, they don't even list currency on their schema.

**Pull request merged**